src/pages/index.js

```diff
@@ -39,38 +42,183 @@ const profFormPopup = new PopupWithForm(
   ".modal_type_profile",
   handleProfileSubmit
 );
+const delCardFormPopup = new PopupWithForm(
+  ".modal_type_delete-card",
+  handleDelCardSubmit
+);
+const editAviFormPopup = new PopupWithForm(
+  ".modal_type_edit-avi",
+  handleEditAviSubmit
+);
 const profUserInfo = new UserInfo({
   userNameSelec: ".profile__name",
   descrSelec: ".profile__title",
+  aviSelec: ".profile__avatar",
+});
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: apiHeaders,
 });
+const delCardApi = new Api({
+  method: "DELETE",
+  headers: apiHeaders,
+});
+
+//INITIALLY SET USER INFO
+api
+  .getUserInfo()
+  .then((data) => {
+    const { name, about, avatar } = data;
+    profUserInfo.setUserInfo({ name, about, avatar });
+  })
+  .catch((err) => console.error(err));
 
 // POPULATE INITIAL CARDS
 export function createCard(data) {
-  const cardElement = new Card(data, "#card-template", handleCardEnlargement);
+  const cardElement = new Card(
+    data,
+    "#card-template",
+    handleCardEnlargement,
+    handleTrashClick,
+    sendLikeReq
+  );
   return cardElement.render();
 }
+api
+  .getCards()
+  .then((data) => {
+    data.reverse();
+    const initialSection = new Section(
+      { items: data, renderer: createCard },
+      ".gallery"
+    );
 
-section.renderItems();
+    initialSection.renderItems();
+  })
+  .catch((err) => console.error(err));
 
 // ENLARGE CARD IMAGE
 function handleCardEnlargement(evt) {
   enlrgImgPopup.open(evt);
 }
 
 // SUBMIT NEW CARD MODAL
-export function handleNewCardFormSubmit(newCardData) {
-  section.addItem(createCard(newCardData));
+function handleNewCardFormSubmit({ name, link }) {
+  const newCardApi = new Api({
+    method: "POST",
+    headers: apiHeaders,
+    body: JSON.stringify({
+      name,
+      link,
+    }),
+  });
 
-  newCardFormPopup.close();
-  newCardModalForm.reset();
+  newCardFormPopup.popup.querySelector(".modal__submit-button").textContent =
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```